lib/include/currentSourceModels.h

```diff
@@ -0,0 +1,114 @@
+#pragma once
+
+// Standard includes
+#include <array>
+#include <functional>
+#include <string>
+#include <tuple>
+#include <vector>
+#include <cmath>
+
+// GeNN includes
+#include "codeGenUtils.h"
+// #include "neuronModels.h"
+#include "newModels.h"
+
+//----------------------------------------------------------------------------
+// Macros
+//----------------------------------------------------------------------------
+#define SET_INJECTION_CODE(INJECTION_CODE) virtual std::string getInjectionCode() const{ return INJECTION_CODE; }
+#define SET_TIME_CONDITION_CODE(TIME_CONDITION_CODE) virtual std::string getTimeConditionCode() const{ return TIME_CONDITION_CODE; }
+#define SET_EXTRA_GLOBAL_PARAMS(...) virtual StringPairVec getExtraGlobalParams() const{ return __VA_ARGS__; }
+
+//----------------------------------------------------------------------------
+// CurrentSourceModels::Base
+//----------------------------------------------------------------------------
+namespace CurrentSourceModels
+{
+//! Base class for all current source models
+class Base : public NewModels::Base
```

Comment:
None of the models you've included use them but ``NewModels::Base`` also includes state variable. I think current sources should definitely have these but you aren't declaring, allocating, initialising or substituting these anywhere! The places to look are generateRunner.cc (I'm afraid that's **REALLY** messy) for declaring and allocating; and generateInit.cc for initialising. In both cases I would recommend doing much the same as the code does for postsynaptic model variables

Reply:
Initially I thought of current sources as code snippets that do not require any variables, which is not always the case...